lib/evmone/instructions.hpp

```diff
@@ -161,21 +161,14 @@ inline void gt(Stack& stack) noexcept
 
 inline void slt(Stack& stack) noexcept
 {
-    // TODO: Move this to intx.
     const auto x = stack.pop();
-    auto& y = stack[0];
-    const auto x_neg = x.hi.hi >> 63;
-    const auto y_neg = y.hi.hi >> 63;
-    y = ((x_neg ^ y_neg) != 0) ? x_neg : x < y;
+    stack[0] = slt(x, stack[0]);
 }
 
 inline void sgt(Stack& stack) noexcept
 {
     const auto x = stack.pop();
-    auto& y = stack[0];
-    const auto x_neg = x.hi.hi >> 63;
-    const auto y_neg = y.hi.hi >> 63;
-    y = ((x_neg ^ y_neg) != 0) ? y_neg : y < x;
+    stack[0] = slt(stack[0], x);
```

Comment:
```suggestion
stack[0] = sgt(stack[0], x);
```

Where are your unit tests? 🙈

Reply:
it was done to limit EVM-origin additions to intx. That would be different if intx had signed integers implementation.
Moreover, from the evmone PoW, it is better to use less functions - this is at least virtually less code. The `LT`/`GT` are implemented using the same pattern. Also because we know that `operator>` is implemented in terms of `operator<` - so we are avoiding some additional indirection.

The proposed change should work performance-wise. The issue is some kind of readability-performance tradeoff.

Reply:
Maybe just leave a comment above "Notice the swapped operands".

Reply:
Single unit test: https://github.com/ethereum/evmone/blob/master/test/unittests/evm_test.cpp#L214.